src/Apps/Order2/Routes/Checkout/CheckoutContext/Order2CheckoutContext.tsx

```diff
@@ -0,0 +1,116 @@
+import { useCheckoutActions } from "Apps/Order2/Routes/Checkout/CheckoutContext/Hooks/useCheckoutActions"
+import { useLoadCheckout } from "Apps/Order2/Routes/Checkout/CheckoutContext/Hooks/useLoadCheckout"
+import type {
+  CheckoutActions,
+  CheckoutState,
+} from "Apps/Order2/Routes/Checkout/CheckoutContext/utils"
+import {
+  CheckoutStepName,
+  CheckoutStepState,
+} from "Apps/Order2/Routes/Checkout/CheckoutContext/utils"
+import type {
+  Order2CheckoutContext_order$data,
+  Order2CheckoutContext_order$key,
+} from "__generated__/Order2CheckoutContext_order.graphql"
+import type React from "react"
+import { createContext, useEffect, useMemo } from "react"
+import { graphql, useFragment } from "react-relay"
+
+export type Order2CheckoutContextValue = CheckoutState & CheckoutActions
+
+export const Order2CheckoutContext =
+  createContext<Order2CheckoutContextValue | null>(null)
+
+interface Order2CheckoutContextProviderProps {
+  order: Order2CheckoutContext_order$key
+  children: React.ReactNode
+}
+
+export const Order2CheckoutContextProvider: React.FC<
+  Order2CheckoutContextProviderProps
+> = ({ order, children }) => {
+  const orderData = useFragment(ORDER_FRAGMENT, order)
+
+  const initialState: CheckoutState = useMemo(
+    () => initialStateForOrder(orderData),
+    [orderData],
+  )
+
+  const { setLoadingError, setLoadingComplete, state, actions } =
+    useCheckoutActions(initialState)
+
+  const isExpressCheckoutLoaded = state.expressCheckoutPaymentMethods !== null
+
+  useLoadCheckout({
+    order: orderData,
+    setLoadingError,
+    setLoadingComplete,
+    isExpressCheckoutLoaded,
+    isLoading: state.isLoading,
+    loadingError: state.loadingError,
+  })
```

Comment:
These lines assemble the checkout context and start the loading process.
- We process the initial state
- We get the usable state and actions from `useCheckoutActions()`
- We start the loading process with the already-existing `useLoadCheckout()` hook, now private to this CheckoutContext directory.

Then below we do the 'pretend express checkout loaded' effect and stick our state + actions into the context provider.

Reply:
If there's already a context present, we shouldn't need to pass arguments to `useLoadCheckout`. Pluck those values out internally to the hook.

Reply:
I'd vote for that. Context seems sufficient!